calcculator1.c

```diff
@@ -1,171 +1,196 @@
+            return 0;
+        }
+
+        
+        if (i > 0 && isOperator(c) && isOperator(exp[i - 1])) 
+           {if(!(c == '-' && !isOperator(exp[i - 2])))   /*Multiple operator should not be there
+                                                            like*+ , but*- will be accepted  */
+           {return 0;}
+           }
+        
+        if (c == '/' && i + 1 < length && exp[i + 1] == '0') {    /* No zero in divisior*/
+            printf("Error: Division by zero \n");
+            return 0;
+        }
+    }
+    return 1;
+}
+
+/* Infix to postfix conversion*/
+char* infixToPostfix(char* exp) {
+    int len = findLen(exp);
+    char* postfix = (char*)malloc((len * 2) * sizeof(char)); 
+    char stack[100];
+    int top = -1;
+    int j = 0;
+
+    for (int i = 0; i < len; i++) {
+        char c = exp[i];
+
+        
+        if ((c == '-' && (i == 0 || isOperator(exp[i - 1]))) || isDigit(c)) { /* if positive or negative digit*/
+            postfix[j++] = c;
+            while (i + 1 < len && isDigit(exp[i + 1])) {
+                postfix[j++] = exp[++i];
+            }
+            postfix[j++] = ' '; /* for space between operator and digit*/
+        } else if (isOperator(c)) {
+            while (top != -1 && precedence(c) <= precedence(stack[top])) {
+                postfix[j++] = stack[top--];
+                postfix[j++] = ' ';
+            }
+            stack[++top] = c;
+        }
+    }
+
+    while (top != -1) {
+        postfix[j++] = stack[top--]; /* if operator left in stack*/
+        postfix[j++] = ' ';
+    }
+
+    postfix[j] = '\0';
+    return postfix;
+}
+
+/* postfix evalution*/
+int postfixEvaluation(char* postfix) {
+    int digitStore[100];
+    int top = -1;
+    int len = findLen(postfix);
+    
+    for (int i = 0; i < len; i++) {
+        char c = postfix[i];
+        if (isDigit(c) || (c == '-' && isDigit(postfix[i + 1]))) {
+            int digit = 0;
+            int sign = 1;
+            if (c == '-') {
+                sign = -1;        /* To handle negative digits*/
+                i++;
+            }
+            while (i < len && isDigit(postfix[i])) {
+                digit = digit * 10 + (postfix[i++] - '0'); /* to handle multiple digits*/
+            }
+            i--; 
+            digitStore[++top] = digit * sign;
+        } else if (isOperator(c)) {
+            int num1 = digitStore[top--];
+            int num2 = digitStore[top--];
+
+            switch (c) {
+                case '+':
+                             digitStore[++top] = num2 + num1; 
+                            break;
+                case '-': 
+                            digitStore[++top] = num2 - num1; 
+                            break;
+                case '*': 
+                            digitStore[++top] = num2 * num1; 
+                            break;
+                case '/': 
+                        if (num1 == 0) {

```

Comment:
Is your program ever reached at this stage, because you mentioned same thing at line no. 85. Correct this, this condition should checked at only one point either here or above